src/ray/core_worker/core_worker.h

```diff
@@ -84,8 +84,39 @@ class CoreWorker {
   // TODO(edoakes): remove this once Python core worker uses the task interfaces.
   void SetCurrentTaskId(const TaskID &task_id) {
     worker_context_.SetCurrentTaskId(task_id);
+    // Clear all actor handles for non-actor tasks.
+    if (actor_id_.IsNil()) {
+      actor_handles_.clear();
+    }
   }
 
+  void SetActorId(const ActorID &actor_id) {
+    RAY_CHECK(actor_id_.IsNil());
+    actor_id_ = actor_id;
+  }
+
+  const ActorID &GetActorId() const { return actor_id_; }
+
+  TaskID GetCallerId() const;
```

Comment:
Comment describing the logic? Especially the embedding an actor ID, that's not very intuitive.